compiler/rustc_infer/src/infer/region_constraints/mod.rs

```diff
@@ -273,7 +272,7 @@ pub(crate) enum UndoLog<'tcx> {
     AddVar(RegionVid),
 
     /// We added the given `constraint`.
-    AddConstraint(Constraint<'tcx>),
+    AddConstraint(usize),
```

Comment:
Could be an IndexVec :)

Reply:
or simply remove the index from `AddConstraint`? :man_shrugging:  It is useless and I only added it for consistency with `AddVerify`.

Reply:
> It is useless and I only added it for consistency with `AddVerify`.
This is wrong. The index was actually needed in in leak_check code, but now I have removed the UndoLog altogether so it doens't matter anymore.

Reply:
Adding a newtype index seems incompatible with sorting and deduplicating the IdexVec IMO.